js/main.js

```diff
@@ -0,0 +1,115 @@
+// to rate limit the api calls
+function throttle(fn, wait) {
+    let timeout = null;
+    return function (...args) {
+        if (!timeout) {
+            timeout = setTimeout(() => {
+                fn.apply(this, args);
+                timeout = null;
+            }, wait);
+        }
+    };
+}
+class Main {
+    // Store fetched data here
+    data = [];
+    isLoading = false; // Track if data is currently being loaded
+    hasMore= true;  // Flag to determine if there's more data to load
+
+    async fetchAndDisplayPosts(page, searchQuery = '') {
+
+        this.isLoading = true;
+        const data = await apiService.fetchPosts(page, 10, searchQuery);
+        this.isLoading = false;
+        this.data = data;
+        this.displayPosts();
+        this.hasMore = data.posts.length > 0; 
+    }
+
+
+    
+    displayPosts() {
+        const searchQuery = document.getElementById('search-input').value;
```

Comment:
create const commonID = { searchInput: 'search-input } and use it. if in-case, if we will change anything in future it will be flexible to change & less change to making human error.

Reply:
done.